jetty-io/src/main/java/org/eclipse/jetty/io/DatagramChannelEndPoint.java

```diff
@@ -0,0 +1,99 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2021 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.io;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import java.nio.ByteBuffer;
+import java.nio.channels.DatagramChannel;
+import java.nio.channels.SelectionKey;
+
+import org.eclipse.jetty.util.BufferUtil;
+import org.eclipse.jetty.util.thread.Scheduler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * <p>An {@link EndPoint} implementation based on {@link DatagramChannel}.</p>
+ */
+public class DatagramChannelEndPoint extends SelectableChannelEndPoint
```

Comment:
Why do we actually need `receive`, `send` and the new `write` on the `EndPoint` interface?
Can't we just write datagram code that depends on this specific `DatagramChannelEndPoint` class for now?
We can abstract away from this once we have experience and other `DatagramEndPoint` implementations